### DISCUSSION TITLE
(rds): Add missing AuroraMysqlEngineVersion version 2.11.1

### DISCUSSION THREAD

**Discussion by ilja-trifonovs-monto** [aws/aws-cdk#24677] [General]
### Describe the feature

AuroraMysqlEngineVersion 2.11.1 is missing in latest CDK

### Use Case

Allow provisioning Aurora 2.11.1 from the CDK

### Proposed Solution

Update AuroraMysqlEngineVersion to include newer versions

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.69.0

### Environment details (OS name and version, etc.)

macOS Ventura Version 13.2.1 (22D68)

**Comment by khushail** [comment]
Hi @ilja-trifonovs-monto , thanks for reaching out. Here is the[ doc link](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.AuroraMysqlEngineVersion.html#static-ver_2_11_1) which shows its already supported by CDK. 
is this what you are looking for?

**Comment by ilja-trifonovs-monto** [comment]
Hi @khushail, yep this is what I am looking for.
I also see in documentation that it is supported, but unfortunately not in real life

`TSError: ⨯ Unable to compile TypeScript:
lib/db/db-cluster.ts:55:87 - error TS2551: Property 'VER_2_11_1' does not exist on type 'typeof AuroraMysqlEngineVersion'. Did you mean 'VER_2_10_1'?

55         engine: DatabaseClusterEngine.auroraMysql({ version: AuroraMysqlEngineVersion.VER_2_11_1 }),`

`$ cdk --version
2.69.0 (build 60a5b2a)`

**Comment by pahud** [comment]
Hi @ilja-trifonovs-monto 

We added this support in https://github.com/aws/aws-cdk/pull/24304

And I can `cdk synth` or `cdk diff` correctly with current cdk version.

<img width="1250" alt="image" src="https://user-images.githubusercontent.com/278432/226000403-305cc8f9-4609-4bca-add3-77beadc9b12b.png">

Let me know if the sample code works with you.

```ts
import * as cdk from 'aws-cdk-lib';
import { aws_rds as rds,
aws_ec2 as ec2 } from 'aws-cdk-lib';
import { Construct } from 'constructs';

export class AuroramysqlStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    new rds.DatabaseCluster(this, 'Cluster', {
      engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_2_11_1 }),
      credentials: rds.Credentials.fromGeneratedSecret('clusteradmin'),
      instanceProps: {
        vpcSubnets: { subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS },
        vpc: ec2.Vpc.fromLookup(this, 'Vpc', { isDefault: true }),
      },
    })
  }
}
```

**Comment by ilja-trifonovs-monto** [comment]
I don't understand why, but when I try to deploy this code I am getting an error

`Iljas-mac:cdk-app iljatrifonovs$ cdk synth CdkAppStack
/Users/iljatrifonovs/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/src/index.ts:843
    return new TSError(diagnosticText, diagnosticCodes, diagnostics);
           ^
TSError: ⨯ Unable to compile TypeScript:
lib/cdk-app-stack.ts:11:93 - error TS2551: Property 'VER_2_11_1' does not exist on type 'typeof AuroraMysqlEngineVersion'. Did you mean 'VER_2_10_1'?

11       engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_2_11_1 }),
                                                                                               ~~~~~~~~~~

  ../../../node_modules/aws-cdk-lib/aws-rds/lib/cluster-engine.d.ts:204:21
    204     static readonly VER_2_10_1: AuroraMysqlEngineVersion;
                            ~~~~~~~~~~
    'VER_2_10_1' is declared here.
lib/cdk-app-stack.ts:14:50 - error TS2339: Property 'PRIVATE_WITH_EGRESS' does not exist on type 'typeof SubnetType'.

14         vpcSubnets: { subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS },
                                                    ~~~~~~~~~~~~~~~~~~~

    at createTSError (/Users/iljatrifonovs/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/src/index.ts:843:12)
    at reportTSError (/Users/iljatrifonovs/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/src/index.ts:847:19)
    at getOutput (/Users/iljatrifonovs/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/src/index.ts:1057:36)
    at Object.compile (/Users/iljatrifonovs/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/src/index.ts:1411:41)
    at Module.m._compile (/Users/iljatrifonovs/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/src/index.ts:1596:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1300:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/iljatrifonovs/.npm/_npx/1bf7c3c15bf47d04/node_modules/ts-node/src/index.ts:1600:12)
    at Module.load (node:internal/modules/cjs/loader:1103:32)
    at Function.Module._load (node:internal/modules/cjs/loader:942:12)
    at Module.require (node:internal/modules/cjs/loader:1127:19) {
  diagnosticCodes: [ 2551, 2339 ]
}

Subprocess exited with error 1`

**Comment by ilja-trifonovs-monto** [comment]
Hi, it is super weird.
for some reason if I put // @ts-ignore before the line with aurora version it works

```
      // @ts-ignore
      engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_2_11_1 }),
```

so I do not understand why visual studio code affects cdk deploy

**Comment by pahud** [comment]
@ilja-trifonovs-monto 

Can you run `npx cdk --version` and `cdk --version` and make sure they are consistent?

I doubt your global cdk CLI executable might be way too old. Can you try `npx cdk synth` again?
